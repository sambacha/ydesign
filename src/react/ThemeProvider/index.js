import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import theme from '../theme';

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {
  theme: PropTypes.object,
};

ThemeProvider.defaultProps = {
  theme,
};

ThemeProvider.displayName = 'ThemeProvider';

export default ThemeProvider;
