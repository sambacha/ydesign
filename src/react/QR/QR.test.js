import QR from './';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

describe('QR component sanity', () => {
  it('has name', () => {
    expect(QR.displayName).toBe('QR');
  });

  it('matches default snapshot', () => {
    const component = render(<QR value="test" />);
    expect(component).toMatchSnapshot();
  });

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <QR value="test" />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
