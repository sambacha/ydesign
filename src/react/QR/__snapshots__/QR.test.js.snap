// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QR component sanity matches default snapshot 1`] = `
<svg
  height="128"
  shape-rendering="crispEdges"
  viewBox="0 0 21 21"
  width="128"
>
  <path
    d="M0,0 h21v21H0z"
    fill="#FFFFFF"
  />
  <path
    d="M0 0h7v1H0zM8 0h2v1H8zM12 0h1v1H12zM14,0 h7v1H14zM0 1h1v1H0zM6 1h1v1H6zM9 1h1v1H9zM12 1h1v1H12zM14 1h1v1H14zM20,1 h1v1H20zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM8 2h1v1H8zM10 2h1v1H10zM12 2h1v1H12zM14 2h1v1H14zM16 2h3v1H16zM20,2 h1v1H20zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM8 3h1v1H8zM11 3h1v1H11zM14 3h1v1H14zM16 3h3v1H16zM20,3 h1v1H20zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM8 4h3v1H8zM14 4h1v1H14zM16 4h3v1H16zM20,4 h1v1H20zM0 5h1v1H0zM6 5h1v1H6zM14 5h1v1H14zM20,5 h1v1H20zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14,6 h7v1H14zM9 7h2v1H9zM0 8h4v1H0zM6 8h1v1H6zM8 8h1v1H8zM10 8h1v1H10zM13 8h1v1H13zM16 8h3v1H16zM20,8 h1v1H20zM0 9h1v1H0zM3 9h3v1H3zM7 9h1v1H7zM10 9h1v1H10zM12 9h1v1H12zM15 9h1v1H15zM17 9h2v1H17zM20,9 h1v1H20zM0 10h1v1H0zM2 10h1v1H2zM4 10h3v1H4zM10 10h1v1H10zM12 10h5v1H12zM19,10 h2v1H19zM0 11h1v1H0zM2 11h2v1H2zM7 11h2v1H7zM10 11h1v1H10zM12 11h2v1H12zM17 11h1v1H17zM19 11h1v1H19zM1 12h2v1H1zM4 12h1v1H4zM6 12h1v1H6zM9 12h3v1H9zM13 12h1v1H13zM16 12h2v1H16zM19 12h1v1H19zM8 13h1v1H8zM10 13h1v1H10zM13 13h1v1H13zM15 13h1v1H15zM17 13h1v1H17zM19 13h1v1H19zM0 14h7v1H0zM10 14h1v1H10zM13 14h2v1H13zM16 14h3v1H16zM0 15h1v1H0zM6 15h1v1H6zM9 15h1v1H9zM11 15h4v1H11zM17 15h2v1H17zM0 16h1v1H0zM2 16h3v1H2zM6 16h1v1H6zM9 16h3v1H9zM14 16h1v1H14zM18,16 h3v1H18zM0 17h1v1H0zM2 17h3v1H2zM6 17h1v1H6zM8 17h2v1H8zM13 17h1v1H13zM16 17h2v1H16zM19 17h1v1H19zM0 18h1v1H0zM2 18h3v1H2zM6 18h1v1H6zM8 18h1v1H8zM10 18h2v1H10zM13 18h1v1H13zM16 18h1v1H16zM18 18h1v1H18zM0 19h1v1H0zM6 19h1v1H6zM8 19h5v1H8zM14 19h1v1H14zM16 19h2v1H16zM20,19 h1v1H20zM0 20h7v1H0zM8 20h2v1H8zM11 20h2v1H11zM15 20h1v1H15z"
    fill="#000000"
  />
</svg>
`;

exports[`QR component sanity matches themed snapshot 1`] = `
<svg
  height="128"
  shape-rendering="crispEdges"
  viewBox="0 0 21 21"
  width="128"
>
  <path
    d="M0,0 h21v21H0z"
    fill="#FFFFFF"
  />
  <path
    d="M0 0h7v1H0zM8 0h2v1H8zM12 0h1v1H12zM14,0 h7v1H14zM0 1h1v1H0zM6 1h1v1H6zM9 1h1v1H9zM12 1h1v1H12zM14 1h1v1H14zM20,1 h1v1H20zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM8 2h1v1H8zM10 2h1v1H10zM12 2h1v1H12zM14 2h1v1H14zM16 2h3v1H16zM20,2 h1v1H20zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM8 3h1v1H8zM11 3h1v1H11zM14 3h1v1H14zM16 3h3v1H16zM20,3 h1v1H20zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM8 4h3v1H8zM14 4h1v1H14zM16 4h3v1H16zM20,4 h1v1H20zM0 5h1v1H0zM6 5h1v1H6zM14 5h1v1H14zM20,5 h1v1H20zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14,6 h7v1H14zM9 7h2v1H9zM0 8h4v1H0zM6 8h1v1H6zM8 8h1v1H8zM10 8h1v1H10zM13 8h1v1H13zM16 8h3v1H16zM20,8 h1v1H20zM0 9h1v1H0zM3 9h3v1H3zM7 9h1v1H7zM10 9h1v1H10zM12 9h1v1H12zM15 9h1v1H15zM17 9h2v1H17zM20,9 h1v1H20zM0 10h1v1H0zM2 10h1v1H2zM4 10h3v1H4zM10 10h1v1H10zM12 10h5v1H12zM19,10 h2v1H19zM0 11h1v1H0zM2 11h2v1H2zM7 11h2v1H7zM10 11h1v1H10zM12 11h2v1H12zM17 11h1v1H17zM19 11h1v1H19zM1 12h2v1H1zM4 12h1v1H4zM6 12h1v1H6zM9 12h3v1H9zM13 12h1v1H13zM16 12h2v1H16zM19 12h1v1H19zM8 13h1v1H8zM10 13h1v1H10zM13 13h1v1H13zM15 13h1v1H15zM17 13h1v1H17zM19 13h1v1H19zM0 14h7v1H0zM10 14h1v1H10zM13 14h2v1H13zM16 14h3v1H16zM0 15h1v1H0zM6 15h1v1H6zM9 15h1v1H9zM11 15h4v1H11zM17 15h2v1H17zM0 16h1v1H0zM2 16h3v1H2zM6 16h1v1H6zM9 16h3v1H9zM14 16h1v1H14zM18,16 h3v1H18zM0 17h1v1H0zM2 17h3v1H2zM6 17h1v1H6zM8 17h2v1H8zM13 17h1v1H13zM16 17h2v1H16zM19 17h1v1H19zM0 18h1v1H0zM2 18h3v1H2zM6 18h1v1H6zM8 18h1v1H8zM10 18h2v1H10zM13 18h1v1H13zM16 18h1v1H16zM18 18h1v1H18zM0 19h1v1H0zM6 19h1v1H6zM8 19h5v1H8zM14 19h1v1H14zM16 19h2v1H16zM20,19 h1v1H20zM0 20h7v1H0zM8 20h2v1H8zM11 20h2v1H11zM15 20h1v1H15z"
    fill="#000000"
  />
</svg>
`;
